src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -124,30 +115,55 @@ public Task<bool> NominateProjectAsync(string projectUniqueName, IVsProjectResto
             }
         }
 
-        private void DumpProjectRestoreInfo(PackageSpec packageSpec)
+        private void DumpProjectRestoreInfo(DependencyGraphSpec projectRestoreInfo)
         {
             try
             {
-                var outputPath = packageSpec.RestoreMetadata.OutputPath;
+                var outputPath = projectRestoreInfo.Projects.First().RestoreMetadata.OutputPath;
                 if (!Directory.Exists(outputPath))
                 {
                     Directory.CreateDirectory(outputPath);
                 }
 
-                // Create dg file
-                var dgFile = new DependencyGraphSpec();
-                dgFile.AddRestore(packageSpec.RestoreMetadata.ProjectName);
-                dgFile.AddProject(packageSpec);
-
                 var dgPath = Path.Combine(outputPath, $"{Guid.NewGuid()}.dg");
-                dgFile.Save(dgPath);
+                projectRestoreInfo.Save(dgPath);
             }
             catch (Exception e)
             {
                 _logger.LogError(e.ToString());
             }
         }
 
+        private static DependencyGraphSpec ToDependencyGraphSpec(ProjectNames projectNames, IVsProjectRestoreInfo projectRestoreInfo)
+        {
+            var dgSpec = new DependencyGraphSpec();
+
+            var packageSpec = ToPackageSpec(projectNames, projectRestoreInfo);
+            dgSpec.AddRestore(packageSpec.RestoreMetadata.ProjectName);
+            dgSpec.AddProject(packageSpec);
+
+            if (projectRestoreInfo.ToolReferences != null)
+            {
+                projectRestoreInfo
+                    .ToolReferences
+                    .Cast<IVsReferenceItem>()
+                    .Select(r => ToToolPackageSpec(projectNames, r))
+                    .ToList()
+                    .ForEach(ts =>
+                    {
+                        dgSpec.AddRestore(ts.RestoreMetadata.ProjectName);
```

Comment:
There is a line above that uses `ProjectName` too. That should have the same fix, right?